Content.Shared/Weapons/Ranged/Systems/SharedGunSystem.cs

```diff
@@ -207,40 +226,52 @@ private void StopShooting(EntityUid uid, GunComponent gun)
     /// <summary>
     /// Attempts to shoot at the target coordinates. Resets the shot counter after every shot.
     /// </summary>
-    public void AttemptShoot(EntityUid user, EntityUid gunUid, GunComponent gun, EntityCoordinates toCoordinates)
+    public void AttemptShoot(EntityUid? user, EntityUid gunUid, GunComponent gun, EntityCoordinates? toCoordinates = null)
     {
-        gun.ShootCoordinates = toCoordinates;
+        if (toCoordinates == null)
+            gun.ShootCoordinates = new EntityCoordinates(gunUid, new Vector2(0, -1)); //Forward vector
+        else
+            gun.ShootCoordinates = toCoordinates;
+
         AttemptShoot(user, gunUid, gun);
         gun.ShotCounter = 0;
     }
 
-    private void AttemptShoot(EntityUid user, EntityUid gunUid, GunComponent gun)
+    private void AttemptShoot(EntityUid? user, EntityUid gunUid, GunComponent gun)
     {
-        if (gun.FireRate <= 0f ||
-            !_actionBlockerSystem.CanUseHeldEntity(user))
+        if (!gun.CanShootWithoutUser && user == null)
+            return;
```

Comment:
Just make another method that treats the gun as the user rather than adding null coalescing.